zebra-chain/src/block/merkle.rs

```diff
@@ -78,6 +83,23 @@ fn hash(h1: &[u8; 32], h2: &[u8; 32]) -> [u8; 32] {
     w.finish()
 }
 
+/// Compute the root of a Merkle tree as used in Bitcoin.
+/// `hashes` must contain the hashes of the tree leaves.
+/// The root is written to the the first element of the input vector.
+/// See [`Root`] for an important disclaimer.
+fn root(hashes: &mut Vec<[u8; 32]>) {
+    while hashes.len() > 1 {
+        *hashes = hashes
+            .chunks(2)
+            .map(|chunk| match chunk {
+                [h1, h2] => hash(h1, h2),
+                [h1] => hash(h1, h1),
```

Comment:
Why write the result to the first element of the input, rather than return it? Maybe [Iterator.fold()](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold) could be used here and return the accumulator.